### PR TITLE
feat: add top progress loading bar

### PR DESCRIPTION
**Pull request by satazor**

### CODE REVIEW

File: docusaurus/docs/what-is-included/top-progress-bar.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+id: top-progress-bar
+title: Top progress bar
+sidebar_label: Top progress bar
+---
+
+A very subtle progress bar is displayed when Next.js is performing route changes, which can take some time depending on how fast the page assets and data (via `getInitialProps`, `getServerSideProps` or `getStaticProps` with `revalidate`).
```
Comment (threequartersjohn) [inline docusaurus/docs/what-is-included/top-progress-bar.md:7]: Small detail, this sentence seems to be incomplete 🤔 
```suggestion
A very subtle progress bar is displayed when Next.js is performing route changes, which can take some time depending on how fast the page assets and data are loaded (via `getInitialProps`, `getServerSideProps` or `getStaticProps` with `revalidate`).
```

Not sure this is the clearest wording to use, but just to exemplify what seems to be missing.

File: docusaurus/docs/what-is-included/top-progress-bar.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+We use the [`NProgress`](https://github.com/rstacruz/nprogress) library for the heavy-lifting of the progress bar UI. Under `shared/modules/nprogress`, you will find a `NProgress` sub-facade that:
+
+- Has `start()` and `done()` methods that ref-count aware, meaning you can call them several times around your code-base; only the first start call and last done call will be accounted to show / hide the progress bar.
```
Comment (threequartersjohn) [inline docusaurus/docs/what-is-included/top-progress-bar.md:11]:
```suggestion
- Has `start()` and `done()` methods that are ref-count aware, meaning you can call them several times around your code-base; only the first start call and last done call will be accounted for to show / hide the progress bar.
```

File: docusaurus/docs/what-is-included/top-progress-bar.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+- Adds a built-in start delay, so that the progress bar is not shown right away.
+
+## Removing this feature
+
+If you are going to have another loading solution for Next.js page transitions, following these steps:
```
Comment (threequartersjohn) [inline docusaurus/docs/what-is-included/top-progress-bar.md:16]:
```suggestion
If you are going to have another loading solution for Next.js page transitions, follow these steps:
```

File: www/shared/styles/global/nprogress.css [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#nprogress {
+    position: fixed;
```
Comment (threequartersjohn) [inline www/shared/styles/global/nprogress.css:2]: Any particular reason not to use our position mixin? Please consider this for the classes below as well.

**Review by threequartersjohn — COMMENTED** [review]
Just some small comments 🙏

**Comment by satazor** [comment]
All done @threequartersjohn, thanks for your awesome review.

**Review by andregoncalvesdev — COMMENTED** [review]
Except the changes that @threequartersjohn suggested regarding the docs/mixin, for me it LGTM.

**Pull request merged**